packages/ra-ui-materialui/src/input/ArrayInput.tsx

```diff
@@ -70,21 +72,37 @@ export const ArrayInput = ({
         ...rest,
     });
 
+    const { touched, error } = fieldProps.meta;
+
     return (
         <FormControl
             fullWidth
             margin="normal"
             className={className}
             {...sanitizeRestProps(rest)}
         >
-            <InputLabel htmlFor={source} shrink>
+            <InputLabel
+                htmlFor={source}
+                shrink
+                variant={variant}
+                error={touched && !!error}
+            >
                 <FieldTitle
                     label={label}
                     source={source}
                     resource={resource}
                     isRequired={isRequired(validate)}
                 />
             </InputLabel>
+            {(touched && error) || helperText ? (
```

Comment:
For consistency shouldn't this be under the items? For other inputs we don't put the helper text between the label and the form control. @fzaninotto what do you think ?